CRM/Export/BAO/ExportProcessor.php

```diff
@@ -2008,8 +2007,9 @@ public function mergeSameAddress() {
           r2.addressee          as master_addressee,
           r2.addressee_id       as master_addressee_id
 FROM      $tableName r1
-INNER JOIN civicrm_address adr ON r1.master_id   = adr.id
-INNER JOIN $tableName      r2  ON adr.contact_id = r2.civicrm_primary_id
+INNER JOIN civicrm_address adr1 ON r1.civicrm_primary_id   = adr1.contact_id AND r1.master_id IS NOT NULL
+INNER JOIN civicrm_address adr2 ON adr2.id = adr1.master_id
+INNER JOIN $tableName      r2  ON adr2.contact_id = r2.civicrm_primary_id
```

Comment:
Due to recent refactoring changes master_id in the exported temp table holds the display name of contact whose address is shared. This is because of formating change in L978 above. Thus this particular query fails, the reason why we used additional INNER JOIN to fetch and JOIN on correct master_id.